source/uwp/Renderer/lib/WholeItemsPanel.cpp

```diff
@@ -47,11 +47,48 @@ AdaptiveNamespaceStart
 
         const Size noVerticalLimit{ availableSize.Width, numeric_limits<float>::infinity() };
 
+        unsigned int stretchableCount{};
         m_visibleCount = count;
-        for (unsigned int i = 0; i < count; i++)
+        for (unsigned int i{}; i < count; ++i)
         {
             ComPtr<IUIElement> spChild;
             RETURN_IF_FAILED(spChildren->GetAt(i, spChild.GetAddressOf()));
+            
+            // If the child is a column it may have stretchable items, so return the column height to auto resize itself
+            ComPtr<IGrid> spChildAsGrid;
+            if (SUCCEEDED(spChild.As(&spChildAsGrid)))
```

Comment:
>if (SUCCEEDED(spChild.As(&spChildAsGrid))) [](start = 12, length = 42)

Does this assume all grids are column sets? That makes me nervous. What happens if we later add more types that use grids?